common/domain/handler.go

```diff
@@ -765,10 +765,10 @@ func (d *handlerImpl) getHistoryArchivalState(
 	}
 	clusterHistoryArchivalConfig := d.archivalMetadata.GetHistoryConfig()
 
-	if requestedConfig != nil && clusterHistoryArchivalConfig.ClusterConfiguredForArchival() {
+	if clusterHistoryArchivalConfig.ClusterConfiguredForArchival() {
 		archivalEvent, err := d.toArchivalUpdateEvent(
-			requestedConfig.HistoryArchivalStatus,
-			requestedConfig.GetHistoryArchivalURI(),
+			updateRequest.HistoryArchivalStatus,
+			updateRequest.GetHistoryArchivalURI(),
 			clusterHistoryArchivalConfig.GetDomainDefaultURI(),
 		)
```

Comment:
took a while to be sure that this would lead to no-ops on nils / empty values... but yep.  looks good.

I feel like there _has_ to be a simpler way to write `archivalConfigStateMachine.go`, but I guess it is at least explicit about covering all possibilities...